components/server/ee/src/workspace/workspace-factory.ts

```diff
@@ -168,6 +174,27 @@ export class WorkspaceFactoryEE extends WorkspaceFactory {
             ws.projectId = project?.id;
             ws = await this.db.trace({ span }).store(ws);
 
+            {
+                const existingPWS = await this.db
+                    .trace({ span })
+                    .findPrebuiltWorkspaceByCommit(
+                        commitContext.repository.cloneUrl,
+                        CommitContext.computeHash(commitContext),
+                    );
+                log.info("createForStartPrebuild–2", {
+                    foundPrebuild: !!existingPWS,
+                    context,
+                    cloneUrl: commitContext.repository.cloneUrl,
+                    commit: CommitContext.computeHash(commitContext),
+                });
+                if (existingPWS) {
```

Comment:
seems to work https://github.com/gitpod-io/gitpod/pull/13113#issuecomment-1255035538